sentry/src/main/java/io/sentry/SentryOptions.java

```diff
@@ -374,8 +380,48 @@ public void addIntegration(@NotNull Integration integration) {
    *
    * @param dsn the DSN
    */
-  public void setDsn(@Nullable String dsn) {
+  public void setDsn(final @Nullable String dsn) {
     this.dsn = dsn;
+
+    dsnHash = calculateHashedDsn(this.dsn);
+  }
+
+  private @Nullable String calculateHashedDsn(final @Nullable String dsn) {
+    if (dsn == null || dsn.isEmpty()) {
+      return null;
+    }
+
+    try {
+      // getInstance() method is called with algorithm SHA-1
+      final MessageDigest md = MessageDigest.getInstance("SHA-1");
```

Comment:
is SHA 256 available in all versions we support? If possible prefer that instead

Reply:
SHA-1 is there from API 1 https://developer.android.com/reference/java/security/MessageDigest
So leaving as it is, the `catch` is needed anyway tho.